roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala

```diff
@@ -242,6 +244,80 @@ object rblstring {
+    * An optional maximum number of splits may be specified
+    * Leading and trailing separators are ignored.
+    * if no speparators, return a Tuple of the string's characters ignoring the count
+    *
+    * examples:
+    * (string-split "a,b,c,d,e,f" ",") ====> ["a" "b" "c" "d" "e" "f" #niv]
+    * (string-split "a,b.c,d.e,f" "," 4) ====> ["a" "b.c" "d.e" "f" #niv]
+    * (string-split "a,b.c,d.e,f" "," 0) ====> []
+    * (string-split "a,b.c,d.e,f" "," -1) ==== []
+    * (string-split ",a,b,c," "," 2) ====> ["a" "b" "c,"]
+    * (string-split ",a,b,c," "," 1) ====> ["a" "b,c,"]
+    * (string-split ",a,b,c," ",") ====> ["a" "b" "c" #niv]
+    * (string-split "a,b,c" "") ====> [#\a #\, #\b #\, #\c]
+    * (string-split "a,b,c,d,e,f" "," 3) ====> ["a" "b" "c" "d,e,f"]
+    * (string-split "a,b.c,d.e,f" "," 2) ====> ["a" "b.c" "d.e,f"]
+    * (string-split "a,b.c,d.e,f" ",." 2) ====> ["a" "b" "c,d.e,f"]
+    * (string-split "a,b.c,d.e,f" "," 1) ====> ["a" "b.c,d.e,f"]
+    */
+  object stringSplit extends Prim {
+    override val name: String = "string-split"
+    override val minArgs: Int = 2
+    override val maxArgs: Int = 3
+
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, Tuple] = {
+      val elem = ctxt.argvec.elem
+
+      for {
+        str <- checkType[RblString](0, elem) // Ensure arg0 is a RblString
+        sep <- checkType[RblString](1, elem) // Ensure arg1 is a RblString
+        w <- checkType[Fixnum](2, elem) // Ensure optional arg2 is a Fixnum
+            match {
+              case Right(v) => Right(v)
+              case Left(_)  => Right(Fixnum(Int.MaxValue))
+            }
+      } yield {
+
+        // At this point:
+        //  str = string to tokenize
+        //  sep = string of separators
+        //  w = count of tokens to return
+
+        val tl = // Get the list of tokens
+          if (sep.value.length > 0)
+            str.value.split(sep.value.toArray) // split ignores trailing separators
+          else
+            "".split(" ") // Don't upset split with an empty list of separators
+
+        val tr = tl.dropWhile(_.length == 0) // Trim off leading separators
+
+        if (sep.value.length == 0) { // no separators? return a Tuple of RblChars
+          Tuple(str.value.toCharArray.toSeq.map(RblChar(_)))
+        } else if (w.value <= 0) { // Invalid count yields an empty Tuple
+          Tuple.NIL
+        } else if (w.value > tr.size) { // Consuming the whole string yields the tokens plus NIV
+          Tuple(tr.toSeq.map(RblString(_)) :+ Ob.NIV)
+        } else {
+          // Handle counts less than available tokens.
+          // This yields the tokens plus the remaining string
+
+          val tokens = tr.slice(0, w.value) // get the selected tokens
+          val idx    = nthIndex(w.value, str.value, sep.value) + 1 // Index to the rest of the string
+          val rest   = str.value.splitAt(idx)._2 // The rest of the string
```

Comment:
`val rest = str.value.drop(idx)` is equivalent and nicer I think than making a tuple with two strings only to throw the first one away.